Objects/iterobject.c

```diff
@@ -427,20 +435,27 @@ anextawaitable_proxy(anextawaitableobject *obj, char *meth, PyObject *arg) {
 
 
 static PyObject *
-anextawaitable_send(anextawaitableobject *obj, PyObject *arg) {
+anextawaitable_send(PyObject *op, PyObject *arg)
+{
+    anextawaitableobject *obj = anextawaitableobject_CAST(op);
     return anextawaitable_proxy(obj, "send", arg);
 }
 
 
 static PyObject *
-anextawaitable_throw(anextawaitableobject *obj, PyObject *arg) {
-    return anextawaitable_proxy(obj, "throw", arg);
+anextawaitable_throw(PyObject *op, PyObject *args)
+{
+    anextawaitableobject *obj = anextawaitableobject_CAST(op);
+    return anextawaitable_proxy(obj, "throw", args);
 }
 
 
 static PyObject *
-anextawaitable_close(anextawaitableobject *obj, PyObject *arg) {
-    return anextawaitable_proxy(obj, "close", arg);
+anextawaitable_close(PyObject *op, PyObject *args)
+{
+    // TODO(picnixz): what happens if we pass an argument to close()?
```

Comment:
args is passed to the proxy, I don't get your TODO.

Reply:
Actually, it's something I didn't understand (which is why I marked it as a TODO):

The `close()` method is marked as:

```c
PyDoc_STRVAR(close_doc,
"close() -> raise GeneratorExit inside generator.")
```

so there is no expected arguments. However, it's marked as 

```c
"close", anextawaitable_close, METH_VARARGS, close_doc
```

So, I wondered why we had `METH_VARARGS` instead of `METH_NOARGS`. But I don't know if we can safely change it to `METH_NOARGS`.

Reply:
You should elaborate the TODO note in this case.

I guess that it's a proxy and so just pass tranparently arguments to the close() method of the wrapped object.

Reply:
I've amended the TODO note and will address it later (or maybe @kumaraditya303 has some idea here)

Reply:
The close method should be `METH_NOARGS`.